src/sidebar/tabs.js

```diff
@@ -3,7 +3,6 @@
 // Selectors that calculate the annotation counts displayed in tab headings
```

Comment:
Me wonders if this should this file be in /sidebar/utils>? I don't have a sense of these things yet, but seems like its a static util type of thing. Thoughts?

Reply:
Looking at this code it's used in rootThread and it seems like there needs to be some cleanup here. I get the impression there is a bit of translation hell going on here and ultimately this whole file will disappear once it's cleaned up.  For example it's currently going from:
rootThread->tabs->annotation-metadata. rootThread is trying to collect all annotations that belong under a certain tab, tabs is calculating per annotation whether an annotation belongs to a certain tab by first determining which tab an annotation belongs to. Metadata does some checks for attributes and things on the annotation to ultimately determine what tab an annotation belongs under. It kinda feels like these should all be functions in the store and when an annotation is added to the store, maybe it should be added with an attribute that identifies in which tab it belongs and maybe we should have a selector where we say give me all annotations under this tab and it returns those or something instead of this jumping around between (as you indicated) what feel like utility files that aren't in the utils area? Long thoughts to have on a PR and something that probably deserves a larger discussion.

Reply:
Speaking of annotation-metadata I'm currently calling into that from the store and after looking at that code I'm wondering if perhaps I should just move that logic into the store right now.

Reply:
> maybe it should be added with an attribute that identifies in which tab it belongs

Generally data in the store follows the principle that it only contains data you couldn't derive from other information already in the store. This avoids cache invalidation bugs where a piece of state is not updated when the state it is derived from changes.

In this context, the tab that an annotation appears in is determined from a) whether it refers to the whole document or a specific part of the document and b) whether it was successfully anchored in the document or not (if (a)) is true.

In future I could imagine that we might make other changes to the tabs and the rules for when annotations appear in them.

> Speaking of annotation-metadata I'm currently calling into that from the store and after looking at that code I'm wondering if perhaps I should just move that logic into the store right now.

As per @LMS007's comment, I'd prefer to put annotation-metadata in the `utils/` directory rather than add the functionality to the store. These are simple functions for answering questions about annotation objects, so from a testing + reusability point of view, I think it is better not to couple them to the store.

The primary constraint on the `util/` directory is that the code should be free of dependencies on other parts of the app (eg. `store/`, `components/`, `services/`).

Reply:
Ok I'll move the annotation-metadata and the tabs into the util area for now. I still feel like these could use some simplification but that will involve root-thread and I'm definitely not going to attempt that right now.